torch_xla/csrc/BUILD

```diff
@@ -0,0 +1,289 @@
+load(
+    "//bazel:rules_def.bzl",
+    "ptxla_cc_library",
+)
+
+genrule(
+    name = "gen_lazy_tensor",
+    srcs = ["//codegen:xla_native_functions.yaml"],
+    outs = [
+        "LazyIr.h",
+        "LazyNonNativeIr.h",
+        "RegisterAutogradXLA.cpp",
+        "RegisterXLA.cpp",
+        "XLANativeFunctions.cpp",
+        "XLANativeFunctions.h",
+    ],
+    cmd = ";".join([
+        "$(location //codegen:lazy_tensor_generator) $(location //codegen:lazy_tensor_generator).runfiles $< $(RULEDIR)",
+        "$(location //codegen:fix_includes) $(OUTS)",
+    ]),
+    tags = [
+        "local",
+        "no-remote-exec",
+    ],
+    tools = [
+        "//codegen:fix_includes",
+        "//codegen:lazy_tensor_generator",
+    ],
+)
+
+ptxla_cc_library(
+    name = "tensor",
```

Comment:
May I ask what's the rules to define each module?

Reply:
Ideally 1 C++ file/header should be 1 library. But this code is written in such a way that everything includes with everything else so that cannot be done as-is. I'll spend some time unwrapping the code in the future to reduce this target. The benefits would be faster incremental compilation, better caching and faster linking time.

Reply:
Yea, it's very unintuitive. Something like grouping with directory makes more sense. haha. Anyway, I guess we can have this cleanup as a follow up.

Reply:
Grouping by directory makes for big slow (re)compilation and linking, with remote caching small nimble objects are the best. I'll do an unspagetti-run some time in the future.

Reply:
I guess I would prefer readability over performance, but that's my own opinion and I haven't used bazel before. Anyway as long as we can provide a general rule and put it as a comment at the top of this file, I'm fine with it.